### PR TITLE
SALT

### PR DESCRIPTION
**Pull request by harmonyn**
Salt now has an overdose threshold. 

The threshold is set at 10, which I believe is higher than any one individual food item, but kicks in if you eat an entire salt shaker.

### CODE REVIEW

**Comment by Supermichael777** [comment]
meme/10

**Comment by optimumtact** [comment]
Yes this is pretty much perfect

**Comment by MrPerson** [comment]
How about we don't make mobcode even worse than it is for a stupid gimmick?

**Comment by Razharas** [comment]
FUn removed, my job here is done

**Comment by Lumipharon** [comment]
Ebin maymays guis :^)

**Comment by Aloraydrel** [comment]
+1 karma

File: code/modules/reagents/Chemistry-Recipes/Others.dm [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
+
 /datum/chemical_reaction/spraytan
 	name = "Spray Tan"
 	id = "spraytan"
 	result = "spraytan"
 	required_reagents = list("orangejuice" = 1, "oil" = 1)
 	result_amount = 2
-	
+
 /datum/chemical_reaction/spraytan2
```
Comment (Dorsisdwarf) [inline code/modules/reagents/Chemistry-Recipes/Others.dm]: Amazing code skills here

Comment (optimumtact) [inline code/modules/reagents/Chemistry-Recipes/Others.dm]: removing excess whitespace is good

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm [hunk 2]
```diff
@@ -237,6 +237,19 @@
...
 	color = "#FFFFFF" // rgb: 255,255,255
+	overdose_threshold = 10
+
+/datum/reagent/consumable/sodiumchloride/overdose_start(mob/living/M)
+	M << "<span class='userdanger'>You feel salty!</span>"
+
+/datum/reagent/consumable/sodiumchloride/overdose_process(mob/living/M)
+	if (!M.salty)
+		M.salty += 3
+	var/salt_message = pick("You feel like punching something.", "You feel fury at everything.", "You feel like raging at the gods.")
```
Comment (Kelenius) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm]: Why not move this line into ```prob()```?

**Comment by phil235** [comment]
Please god no. We're not adding a var to all mobs and another check to life() just for a silly meme.

**Comment by Miauw62** [comment]
this code is shit.
sorry, but no.

**Comment by optimumtact** [comment]
>this code is shit

no it's not, it's objectively as good as the other speech affectors

**Comment by Miauw62** [comment]
Yes, it's a well known fact that a check that passes once every 500 times
is just as acceptable as one that passes every 5000000 times, right?
Op 15-sep.-2015 23:46 schreef "Orange Borg" <notifications@github.com>:

**Comment by optimumtact** [comment]
The hell, if you have a problem with how the speech code works feel free to rewrite it a second time

**Comment by optimumtact** [comment]
and I'd love to see your solution that is faster than a single if boolean check

**Comment by Lumipharon** [comment]
>var/salt_message = pick("You feel like punching something.", "You feel fury at everything.", "You feel like raging at the gods.")

>not a meme PR

**Comment by Dorsisdwarf** [comment]
I liked it better when it was complaining about security/ the captain/ the AI because spessmens actually do that.

**Comment by lordpidey** [comment]
Optimumtact, take a look at how spray tan's overdose works.

It has annoying messages pop up, but doesn't add another check to every single mob.

**Comment by chimerasable** [comment]
Useless junk.

**Comment by bgobandit** [comment]
ok, all joiks aside, what needs to be done to fix this

**Comment by Razharas** [comment]
I dunno its just not funny and the code it light enough for the lame joke to pass unnoticably

**Comment by bgobandit** [comment]
how about if I fix the capitalization code which seems to be the main issue and rewrite the responses

**Comment by Aranclanos** [comment]
life() is already too toggled to be adding gimmicks like this, the var/salty is unacceptable

**Comment by Bawhoppen** [comment]
Bring this back please

**Pull request closed without merging**